src/dmd/clone.d

```diff
@@ -970,6 +970,12 @@ DtorDeclaration buildDtor(AggregateDeclaration ad, Scope* sc)
     switch (ad.dtors.dim)
     {
     case 0:
+        /* If running invariant, need a destructor to hang it on,
+         * but only do for root modules, as ones in the library may not
+         * have been compiled with useInvariants
```

Comment:
Contracts (and possibly invariants) should really be generated and called by the caller, just like a template.